velox/connectors/hive/storage_adapters/s3fs/tests/S3FileSystemTest.cpp

```diff
@@ -224,3 +209,50 @@ TEST_F(S3FileSystemTest, invalidCredentialsConfig) {
     }
   }
 }
+
+TEST_F(S3FileSystemTest, missingFile) {
+  const char* bucketName = "data1";
+  const char* file = "i-do-not-exist.txt";
+  const std::string s3File = s3URI(bucketName, file);
+  addBucket(bucketName);
+  auto hiveConfig = minioServer_->hiveConfig();
+  filesystems::S3FileSystem s3fs(hiveConfig);
+  s3fs.initializeClient();
+  try {
+    s3fs.openFileForRead(s3File);
+    FAIL() << "Expected VeloxException";
+  } catch (VeloxException const& err) {
+    EXPECT_EQ(
+        err.message(),
+        std::string(
+            "Failed to get metadata for S3 object due to: 'Resource not found'. Path:'s3://data1/i-do-not-exist.txt', SDK Error Type:16, HTTP Status Code:404, S3 Service:'MinIO', Message:'No response body.'"));
+  }
+}
+
+TEST_F(S3FileSystemTest, invalidAccessKeySecretKey) {
+  const std::unordered_map<std::string, std::string> config({
```

Comment:
I wonder if it would be handy to modify MinioServer::hiveConfig method to accept an optional map of config overrides, e.g.

```
auto hiveConfig = minioServer_->hiveConfig({"hive.s3.aws-access-key", "dummy-key"});
```

or

```
auto hiveConfig = minioServer_->hiveConfig({"hive.s3.aws-secret-key", "dummy-key"});
```

It would also be nice to test invalid access and secret keys separately.

Reply:
Yeah! Will do.